azurerm/resource_arm_servicebus_rule_test.go

```diff
@@ -0,0 +1,322 @@
+		reply_to_session_id = "test_reply_to_session_id"
+		content_type        = "test_content_type"
+	  }
+	`
+	return fmt.Sprintf(testAccAzureRMServiceBusRule_tfTemplate, rInt, location, rInt, rInt, rInt, rInt, correlationFilter)
+}
+
+func testAccAzureRMServiceBusRule_correlationFilter(rInt int, location string) string {
+	correlationFilter := `
+	filter_type = "CorrelationFilter"
+	correlation_filter = {
+		correlation_id      = "test_correlation_id"
+		message_id          = "test_message_id"
+	  }
+	`
+	return fmt.Sprintf(testAccAzureRMServiceBusRule_tfTemplate, rInt, location, rInt, rInt, rInt, rInt, correlationFilter)
+}
+
+func testAccAzureRMServiceBusRule_correlationFilterUpdated(rInt int, location string) string {
+	correlationFilter := `
+	filter_type = "CorrelationFilter"
+	correlation_filter = {
+		correlation_id      = "test_correlation_id"
+		message_id          = "test_message_id_updated"
+		reply_to            = "test_reply_to_added"
+	  }
+	`
+	return fmt.Sprintf(testAccAzureRMServiceBusRule_tfTemplate, rInt, location, rInt, rInt, rInt, rInt, correlationFilter)
+}
+
+func testAccAzureRMServiceBusRule_correlationFilterWithAction(rInt int, location string) string {
+	correlationFilter := `
+	filter_type = "CorrelationFilter"
+	correlation_filter = {
+		correlation_id      = "test_correlation_id"
+		message_id          = "test_message_id"
+	  }
+	  action      = "SET Test='true'"
+	`
+	return fmt.Sprintf(testAccAzureRMServiceBusRule_tfTemplate, rInt, location, rInt, rInt, rInt, rInt, correlationFilter)
+}
+
+const resourceName = "azurerm_servicebus_rule.test"
+
+const testAccAzureRMServiceBusRule_tfTemplate = `
+resource "azurerm_resource_group" "test" {
+    name     = "acctestRG-%d"
+    location = "%s"
+}
+
+resource "azurerm_servicebus_namespace" "test" {
+    name                = "acctestservicebusnamespace-%d"
+    location            = "${azurerm_resource_group.test.location}"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    sku                 = "standard"
+}
+
+resource "azurerm_servicebus_topic" "test" {
+    name                = "acctestservicebustopic-%d"
+    namespace_name      = "${azurerm_servicebus_namespace.test.name}"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_servicebus_subscription" "test" {
+    name                = "acctestservicebussubscription-%d"
+    namespace_name      = "${azurerm_servicebus_namespace.test.name}"
+    topic_name          = "${azurerm_servicebus_topic.test.name}"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    max_delivery_count  = 10
+}
+
+resource "azurerm_servicebus_rule" "test" {
+	name                = "acctestservicebusrule-%d"
+	namespace_name      = "${azurerm_servicebus_namespace.test.name}"
+	topic_name          = "${azurerm_servicebus_topic.test.name}"
+	subscription_name   = "${azurerm_servicebus_subscription.test.name}"
+	resource_group_name = "${azurerm_resource_group.test.name}"	
+	%s
```

Comment:
can we duplicate this into each test and make the spacing consistent here?

Reply:
Is there any reason to duplicate, the topic, subscription and resource group as these will be the same for each test

Reply:
Ok sure, i get what you mean